types/sharedb/lib/client.d.ts

```diff
@@ -1,9 +1,11 @@
 /// <reference path="sharedb.d.ts" />
 import * as WS from 'ws';
 import * as ShareDB from './sharedb';
+import Agent = require('./agent');
 
 export class Connection {
     constructor(ws: WebSocket | WS);
+    agent?: Agent;
```

Comment:
The `agent` property [is always set](https://github.com/share/sharedb/blob/v1.1.0/lib/client/connection.js#L58-L61), with a default value of `null`. For accuracy, I'd suggest:

```suggestion
    agent: Agent | null;
```

It'd also be good to add a JSDoc comment indicating that only same-process server connections will have an agent set.